src/cmap/errors.ts

```diff
@@ -39,6 +40,22 @@ export class PoolClearedError extends MongoNetworkError {
   }
 }
 
+/**
+ * An error indicating that a connection pool has been cleared after the monitor for that server timed out.
+ * @category Error
+ */
+export class PoolClearedOnNetworkError extends PoolClearedError {
+  constructor(pool: ConnectionPool) {
+    super(pool, `Connection to ${pool.address} interrupted due to server monitor timeout`);
+
+    this.addErrorLabel(MongoErrorLabel.RetryableWriteError);
```

Comment:
we should not do this here because this error already inherits its retryability logic from the `PoolClearedError`

Reply:
Per earlier discussion, let's add the label in `PoolClearedError` and remove the manual adding of the label in `withConnection`, and also get rid of the `isRetryableWritesEnabled` and the corresponding gate in `makeOperationHandler` entirely since `executeOperation` already has the `willRetryWrite` (and read) logic

Reply:
https://jira.mongodb.org/browse/NODE-4838 relevant ticket